include/mbgl/style/style_properties.hpp

```diff
@@ -169,22 +165,16 @@ class BackgroundPaintProperties {
     float opacity = 1.0f;
     Color color = {{ 0, 0, 0, 1 }};
     Faded<std::string> image;
+
+    inline bool isVisible() const {
+        return opacity > 0;
+    }
 };
```

Comment:
What about adding a `PaintProperties` that every `____PaintProperties` inherits containing `isVisible` as a pure virtual?